lambda_layers/core_layer/python/core_layer/handler/user_handler.py

```diff
@@ -122,11 +122,90 @@ def get_top_users(n, attr, descending, is_test, session) -> [User]:
     users: [User]
         A list including the top n user objects as ordered by attr, desc
     """
-    session = get_db_session(is_test, session)
+
+    if session == None:
+        session = get_db_session(is_test, session)
+
     sort_column = getattr(User, attr).desc() if descending else getattr(User, attr)
+    
     users = session.query(User).order_by(sort_column).limit(n).all()
     return users
 
+
+def get_top_users_by_period(n, p, attr, descending, is_test, session) -> [User]:
+    """
+    Returns the top "n" users in a period (1 week)
+    sorted by "attr" in descending or ascending order as set by "descending"
+
+    Parameters
+    ----------
+    n: int, required
+        the number of users to return
+    p: period weeks, required
+        
+    attr: str, required
+        the column on the users table to sort by
+    descending: bool, required
+        which order to sort the rows by column 'attr' in False = ASC or True =DESC
+    is_test: bool, required
+        is this code being run as part of the tests or in production
+    session: Session??, required
+        a database session
+    Returns
+    ------
+    users: [User]
+        A list including the top n user objects as ordered by attr, desc
+    """
+     
+    if session == None:
+        session = get_db_session(is_test, session)
+
+    compare_timestamp = datetime.now() - timedelta(weeks=p)
```

Comment:
For this to work with our database, you need to use `helper.get_date_time(datetime.now() - timedelta(weeks=p), is_test)`. This is because the test db engine (SQLite) works with datetime objects, while our AWS DB works with date strings.

Reply:
Done. 
Works with `helper.get_date_time(datetime.now(), is_test) - timedelta(weeks=p)`